public/components/query_compare/home.tsx

```diff
@@ -33,33 +39,110 @@ export const Home = ({
   setBreadcrumbs,
   setToast,
   chrome,
+  savedObjects,
+  dataSourceEnabled,
+  dataSourceManagement,
+  setActionMenu,
 }: QueryExplorerProps) => {
   const {
-    documentsIndexes,
-    setDocumentsIndexes,
-    pipelines,
-    setPipelines,
     showFlyout,
+    documentsIndexes1,
+    documentsIndexes2,
+    setDocumentsIndexes1,
+    setDocumentsIndexes2,
+    datasource1,
+    datasource2,
+    setFetchedPipelines1,
+    setFetchedPipelines2,
   } = useSearchRelevanceContext();
 
   useEffect(() => {
     setBreadcrumbs([...parentBreadCrumbs]);
   }, [setBreadcrumbs, parentBreadCrumbs]);
+  const [dataSourceOptions, setDataSourceOptions] = useState<DataSourceOption[]>([]);
+  const fetchIndexes = (dataConnectionId: string, queryNumber: string) => {
+    if(dataConnectionId){
+      http.get(`${ServiceEndpoints.GetIndexes}/${dataConnectionId}`).then((res: DocumentsIndex[]) => {
+            if(queryNumber == QUERY_NUMBER_ONE){
+              setDocumentsIndexes1(res)
+            }
+            else{
+              setDocumentsIndexes2(res)
+            }
+          });
+    }
+    else{
+      http.get(ServiceEndpoints.GetIndexes).then((res: DocumentsIndex[]) => {
+        if(queryNumber == QUERY_NUMBER_ONE){
+          setDocumentsIndexes1(res)
+        }
+        else{
+          setDocumentsIndexes2(res)
+        }
+      })
+    }
+  }
+  const fetchPipelines = (dataConnectionId: string, queryNumber: string) => {
+    if(dataConnectionId){
+      http.get(`${ServiceEndpoints.GetPipelines}/${dataConnectionId}`).then((res: {}) => {
+        if(queryNumber == QUERY_NUMBER_ONE){
+          setFetchedPipelines1(res)
+        }
+        else{
+          setFetchedPipelines2(res)
+        }
+      });
+    }
+    else{
+      http.get(ServiceEndpoints.GetPipelines).then((res: {}) => {
+        if(queryNumber == QUERY_NUMBER_ONE){
+          setFetchedPipelines1(res)
+        }
+        else{
+          setFetchedPipelines2(res)
+        }
+      })
+    }
+  }
```

Comment:
May be we can have an another way to write this part as it contains similar code？
```
const fetchData = (endpoint: string, setMethod: Function, queryNumber: string, dataConnectionId?: string) => {
  
  const url = dataConnectionId ? `${endpoint}/${dataConnectionId}` : endpoint;
  
  http.get(url).then((res: any) => {
    if (queryNumber === QUERY_NUMBER_ONE) {
      setMethod(res);
    } else {
      setMethod2(res);
    }
  });
};

const fetchIndexes = ....

const fetchPipelines = ...
 ```